src/mame/act/victor9k.cpp

```diff
@@ -312,14 +323,35 @@ MC6845_UPDATE_ROW( victor9k_state::crtc_update_row )
 	}
 }
 
+MC6845_BEGIN_UPDATE( victor9k_state::crtc_begin_update )
+{
+	uint16_t ma = m_crtc->get_ma();
+	int hires = BIT(ma, 13);
+	int width = hires ? 16 : 10;
+	if (hires != m_hires) {
+		LOGDISPLAY("mc6845 begin update change resolution: %s\n", hires ? "high" : "low");
+		if (hires) 
+		{
+			m_screen->set_raw(15_MHz_XTAL / 5 , 1488, 0, 1279, 422, 0, 410);
+			m_screen->set_visible_area(0,1279,0,410);
+		} else
+		{
+			m_screen->set_raw(15_MHz_XTAL / 5 , 930, 0, 799, 422, 0, 410);
+			m_screen->set_visible_area(0,799,0,410);
+		}
+		m_crtc->set_hpixels_per_column(width);
+		m_crtc->set_char_width(width);
+		m_crtc->set_visarea_adjust(0, 0, 0, 10);  //show line 25
+		m_hires = hires;
+	}
```

Comment:
This file uses Allman formatting – please stick with that, don’t use a weird mix of Allman and K&R.

Also, this can’t be right.  You don’t use `set_raw` outside machine configuration member functions, and you don’t use `set_visible_area` with `set_raw` as it’s implied by the first visible/blanking pixel counts.  You use `configure` to reconfigure the screen dynamically.  You shouldn’t be using hard-coded numbers like this either, because the parameters are programmed into the CRTC by the software.

Have you looked at other systems in MAME that use a 6845?

Reply:
OK, so at this point I've removed the extra calls to set_raw. I believe I'm utilizing the MC6845 as intended. The original problem I ran into is back, with the bottom of the screen loosing roughly 10 pixels. Is there a clean way to handle that so these pixels render correctly? Or should I be looking into a bug in the 6845 implementation?

Reply:
`set_border_area(true)` makes me think that the 6845 is setting up the screen into strips, is it trying to change 6845 values on the fly?

Reply:
I'm not sure what you mean by on the fly. The system supports two resolution modes and changes them dynamically based on software.

Reply:
I checked in local build, it changes the cursor addresses on different scanline positions, which may be for a split effect (and I can't trust the timings).
PS: it also make MAME to crash if I leave pinging `wpset 0xe8000,2,w`, which is definitely something not supposed to happen and hopefully fixed in the CI build provided here.

Reply:
I was able to run a local build with `wpset 0xe8000,2,w` pinging in the debugger.